ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -164,9 +155,53 @@ type ModernLedgerDB l = DbChangelog l
 -- blocks. For example, if we are on line (*), and roll back 6 blocks, we get
 --
 -- > L3 |> []
+type LegacyLedgerDB l = AnchoredSeq
+                               (WithOrigin SlotNo)
+                               (Checkpoint (l ValuesMK))
+                               (Checkpoint (l ValuesMK))
+
+-- | Type alias for the new-style database
+--
+-- This type is internally a DbChangelog which as an example, with @k = 3@ could look like:
+--
+-- > stateAnchor   | diskAnchor | states                     | tableDiffs
+-- > ----------------------------------------------------------------------------
+-- >      0        |      0     | [ L0]                     | [ ]
+-- >      0        |      0     | [ L0, L1]                 | [ D1]
+-- >      0        |      0     | [ L0, L1, L2]             | [ D1, D2]
+-- >      0        |      0     | [ L0, L1, L2, L3]         | [ D1, D2, D3]
+-- >      1        |      0     | [ L0, L1, L2, L3, L4]     | [ D1, D2, D3, D4]
+-- >      2        |      0     | [ L0, L1, L2, L3, L4, L5] | [ D1, D2, D3, D4, D5]    (*)
+-- >      2        |      2     | [ L2, L3, L4, L5]         | [ D3, D4, D5]   -- flush (**)
+-- >      3        |      2     | [ L2, L3, L4, L5, L6]     | [ D3, D4, D5, D6]
+--
+-- The disk anchor moves when we flush data to the disk, and the state anchor
+-- points always to the state that represents the tip of the logical immutable
+-- database. Notice that @seqNo (last states) - stateAnchor@ is usually @k@
+-- (unless rollbacks or corruption take place). We cannot roll back more than
+-- @k@ blocks, this means that under a rollback of @k@ blocks at (*), the
+-- database will look something like this:
+--
+-- >      2        |      0     | [ L0, L1, L2]             | [ D1, D2]
+--
+-- And a rollback of @k@ blocks at (**) will look something like this:
+--
+-- >      2        |      0     | [ L2]                     | [ ]
+--
+-- Notice how the states list always contains the in-memory state of the anchor,
+-- but the table differences might not contain the differences for that anchor
+-- if they have not been flushed to the backend.
+--
+-- The new-style database has to be coupled with the @BackingStore@ which
```

Comment:
I wonder if this comment is relevant here: there's no reference to the backing store, or to the pointers to the on-disk data.

Reply:
I think it is relevant. The backing store is used in this module, and although not mentioned here, it has to be known and understood by the reader in order to use this module. And I don't think we should explain here what it is, that should live within the BackingStore definition I think